### PR TITLE
Gateway api design

### PR DESCRIPTION
**Pull request by Arslanka**

### CODE REVIEW

File: api.yaml [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+        401:
+          $ref: '#/components/responses/401'
+        404:
+          $ref: '#/components/responses/404'
+
+  /v1/booking/reservation/{reservationId}:
+    get:
+      operationId: GetReservationRequest
+      tags:
+        - Users
```
Comment (kadeezyy) [inline api.yaml:153]: точно ли правильный тег?

Comment (Arslanka) [reply api.yaml:153]: не точно =)

File: api.yaml [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+          type: object
+          properties:
+            id:
+              type: string
+            bookId:
+              type: string
+            userId:
+              type: string
+            reservationStatus:
+              type: string
```
Comment (vityaman) [inline api.yaml:396]: You forgot `required` section maybe?

File: api.yaml [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+      type: object
+      properties:
+        id:
+          type: string
+        title:
+          type: string
+        author:
+          type: string
+        content:
+          type: string
```
Comment (vityaman) [inline api.yaml:417]: You include the whole book content? Are we booking service for libraries or an online library?

File: api.yaml [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+          items:
+            $ref: '#/components/schemas/BookContentResponse'
+        currentPage:
+          type: integer
+        pageSize:
+          type: integer
+        totalPages:
+          type: integer
+        totalItems:
+          type: integer
```
Comment (vityaman) [inline api.yaml:433]: Total pages can be calculated on the frontend from `totalItems` and `pageSize`.

**Review by vityaman — CHANGES_REQUESTED** [review]

**Pull request merged**